update kfserving website for 0.4.1 release (#2342)
kubeflow/website: content/en/docs/components/serving/kfserving.md

* Enable a simple, pluggable, and complete story for your production ML inference server by providing prediction, pre-processing, post-processing and explainability out of the box.

> Our strong community contributions help KFServing to grow. We have a Technical Steering Committee driven by AWS, IBM, Nvidia, Seldon, and Bloomberg. [Browse the KFServing GitHub repo](https://github.com/kubeflow/kfserving) to give us feedback!

Reviewer: Just a suggestion - official unofficial names maybe:

```suggestion
Our strong community contributions help KFServing to grow. We have a Technical Steering Committee driven by Amazon Web Services (AWS), IBM Cloud, NVIDIA, Seldon, and Bloomberg. [Browse the KFServing GitHub repo](https://github.com/kubeflow/kfserving) to give us feedback!
```

(Had to go [here](https://s22.q4cdn.com/364334381/files/doc_financials/2020/ar/2020-nvidia-annualreport-content-r25-web-144dpi-combined.pdf) to check "NVIDIA" 🤣 Their site uses all caps everywhere, I think)

## Install with Kubeflow

KFServing works with Kubeflow 1.2. Kustomize installation files are [located in the manifests repo](https://github.com/kubeflow/manifests/tree/master/kfserving).
See examples running KFServing on [Istio/Dex](https://github.com/kubeflow/kfserving/tree/master/docs/samples/istio-dex). For installation on major cloud providers with Kubeflow, please follow their installation docs. 

> Kubeflow 1.2 includes KFServing v0.4.1, where the focus has been on enabling KFServing on OpenShift, and additionally providing more features like adding batcher module as sidecar, Triton inference server renaming and integrations, upgrading Alibi explainer to 0.4.0, allowing customized url paths on data plane, updating logger to CloudEvents V1 protocol etc. Additionally minimum Istio version has been moved to Istio v1.3.1. More details can be found [here](https://github.com/kubeflow/kfserving/releases/tag/v0.4.0) and [here](https://github.com/kubeflow/kfserving/releases/tag/v0.4.1)

Reviewer: ```suggestion
Kubeflow 1.2 includes KFServing v0.4.1, where the focus has been on enabling KFServing on OpenShift and additionally providing more features, such as adding batcher module as sidecar, Triton inference server renaming and integrations, upgrading Alibi explainer to 0.4.0, and allowing customized URL paths on data plane, updating logger to CloudEvents V1 protocol. Additionally, the minimum Istio version is now v1.3.1. More details can be found [here](https://github.com/kubeflow/kfserving/releases/tag/v0.4.0) and [here](https://github.com/kubeflow/kfserving/releases/tag/v0.4.1)
```

WDYT
